### DISCUSSION TITLE
Add Array method to move item to front or back

### DISCUSSION THREAD

**Discussion by maxbobse** [godotengine/godot-proposals#10093] [Scripting]
I think it would be nice to have a method that allows me to move an item of an array to front or rear of the array. In my use case I have an array with values [1, 2, 3] (or any other variant items) and loop over it round-robin so it changes

[1, 2, 3]
[2, 3, 1]
[3, 1, 2]
[1, 2, 3]
and so on ...

Right now I am using remove() and append() to achieve my goal but it would be nice to reduce it to a one-liner instead. The already existing push_back() and push_front() are actually creating a copy of the value and pushing them to the end of the array instead of moving it thus increasing the size of the array.

**Comment by AThousandShips** [comment]
A simple one-liner would be just `arr.push_back(arr.pop_at(i))`

**Reply by maxbobse** [reply]
Rather
arr.push_back(arr.pop_at(arr.find(myItem)))

and yes, its a one-liner but not really convenient to use.

**Reply by AThousandShips** [reply]
If you want to access a specific value yes, but the method shouldn't probably be a by value but by index

If you know what the value at an index is this is completely unnecessary IMO, if anything it should be a move by index, but doing it by value is very specific and unhelpful in general, accessing by index is far more general and useful

**Reply by maxbobse** [reply]
Then there should be two methods: one by value and one by index.

In my scenarios my input criteria is the value so even if I would want to use the index for the new method I'd always have to run the find() first anyways.
